app/javascript/packs/body.jsx

```diff
@@ -0,0 +1,183 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import PropTypes from 'prop-types'
+import ReactPaginate from 'react-paginate';
+import Timelogs from './timelogs.jsx'
+import NewTimelog from './new_timelog.jsx'
+
+class Body extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = { timelogs: [], trello_cards: [], page: 0, pageCount: 1, perPage: 3, start_time: 0, end_time: 0 };
+    this.handleSubmit = this.handleSubmit.bind(this, this.state);
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.removeTimelog = this.removeTimelog.bind(this, this.state);
+    this.updateTimelogs = this.updateTimelogs.bind(this, this.state);
+    this.loadTimelogs = this.loadTimelogs.bind(this, this.state);
+    this.filterByDuration = this.filterByDuration.bind(this, this.state);
+    this.filterByStartTime = this.filterByStartTime.bind(this, this.state);
+    this.filterByEndTime = this.filterByEndTime.bind(this, this.state);
+    this.filterByTimeRange = this.filterByTimeRange.bind(this, this.state);
```

Comment:
To bind only `this` in most cases is enough. Did you have any problems that you've decided to bind `this.state` as well?